telemetry/internal/appinsights/transmitter.go

```diff
@@ -138,30 +138,20 @@ func canRetryBackendError(berror contracts.BackendResponseError) bool {
 		berror.StatusCode == tooManyRequestsOverExtendedTimeResponse
 }
 
-func (result *transmissionResponse) getRetryItems(items []batchItem) (succed, failed int, retries []batchItem) {
-	defer func() {
-		for i := len(retries) - 1; i >= 0; i-- {
-			item := &retries[i]
-			// If the item has been retried too many times, we consider it failed.
-			if item.retries > 2 {
-				retries = retries[:i]
-				failed++
-				continue
-			}
-			item.retries++
-		}
-	}()
-	if result.statusCode == partialSuccessResponse {
+// result returns the number of succeeded and failed items, and a list of items that can be retried.
+// Items is the complete list of result that was sent.
+func (resp *transmissionResponse) result(items []batchItem) (succeed, failed int, retries []batchItem) {
```

Comment:
Ah, I think I get what you're going for with `succed` now. It normally reads to me as simply a misspelling or as "sucked" with unusual spelling (maybe just a US slang thing?), but I think I've seen this as `succCount` (and `failCount`) before elsewhere. I think that would be readable enough with the context of `Count`.

(Could also go with `succeeded` (like the comment) to fit tense with `failed`, but I get abbreviating.)

I think `retries` would also be clearer as `retryItems` like it's actually named at the call site--these aren't retries yet, just items that can/should be retried. Perhaps `rs` is a fine abbreviation that naturally directs more attention to the doc comment where it's spelled out (and the call site can keep its longer name for the local var).

Reply:
Thanks, make sense!